spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/operations/SqsOperations.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs.operations;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Consumer;
+import org.springframework.messaging.Message;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchResponse;
+
+/**
+ * Sqs-specific synchronous messaging operations for Standard and Fifo queues.
+ * <p>
+ * Note that the Standard queue methods can be used for Fifo queues as long as necessary headers are added for required
+ * attributes such as message deduplication id. See {@link io.awspring.cloud.sqs.listener.SqsHeaders} for reference of
+ * available headers.
+ * <p>
+ * Fifo queue methods accept the required attributes and add a random value if none is specified.
+ *
+ * @author Tomaz Fernandes
+ * @since 3.0
+ * @param <T> the payload type
+ */
+public interface SqsOperations<T> extends MessagingOperations<T> {
+
+	/**
+	 * Send a message to a Standard SQS queue using the {@link SqsSendOptions.Standard} options.
+	 * @param to a {@link SqsSendOptions.Standard} consumer.
+	 * @return The {@link UUID} of the message.
+	 */
+	SendResult<T> send(Consumer<SqsSendOptions.Standard<T>> to);
+
+	/**
+	 * Send a message to a Fifo SQS queue using the {@link SqsSendOptions.Fifo} options.
+	 * @param to a {@link SqsSendOptions.Fifo} consumer.
+	 * @return The {@link UUID} of the message.
+	 */
+	SendResult<T> sendFifo(Consumer<SqsSendOptions.Fifo<T>> to);
```

Comment:
Could we merge it into single method? By having this separating to `sendFifo` and `send` it gets unclear how other  methods can be used. For example: `SendResult<T> send(T payload);` - is it only for standard queue, fifo or both? It's quite impossible to tell without getting into details.

Reply:
Yeah, I was in doubt about this. The main point of having them separate would be users explicitly selecting the FIFO methods and allowing them to add FIFO-specific properties, while cleaning up the regular `send` API. Also, we add the `messageGroupId` and `messageDeduplicationId` on the user's behalf if it's not specified.

But perhaps it's doing more harm than good.

I'll merge the FIFO-specific properties to the send method, and perhaps we can detect it's a `fifo` queue and add the missing properties if none is specified.